### PR TITLE
Bookstore: add reducers and actions

### PR DESCRIPTION
**Pull request by amejid**
## Milestone 2: Add reducers and actions

In this PR, the following were implemented:-

- No linter errors
- Followed the correct Gitflow
- Created redux store and reducers for books and categories

### CODE REVIEW

File: src/redux/books/books.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    removeBook(state, action) {
+      const foundId = state.books.findIndex((book) => book.id === action.payload);
+      state.books.splice(foundId, 1);
+    },
+  },
+});
+
+export const booksActions = booksSlice.actions;
+
+export default booksSlice.reducer;
```
Comment (Bondok6) [inline src/redux/books/books.js]: - [ ] I can see that you used `createSlice` from `toolkit` to write your book's actions and reducers and this is not wrong, But according to the requirements, you should write your book's actions and reducer using the [ducks pattern](https://github.com/erikras/ducks-modular-redux). So kindly recreate this file using [ducks pattern](https://github.com/erikras/ducks-modular-redux) to follow the requirements 💯👌

> It should be something like

```
// Define Action Type
const ADD = 'bookstore/books/ADD';
const REMOVE = 'bookstore/books/REMOVE';

export default function booksReducer(state = [], action) {
  switch (action.type) {
    case ADD: return ... ;
    case REMOVE: return ... ;
    default: return state;
  }
}

export function addBook(book) { ... }

export function removeBook(id) { ... }
```

File: src/redux/categories/categories.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  reducers: {
+    checkStatus() {
+      return 'Under Construction';
+    },
+  },
+});
+
+export const categoriesActions = categoriesSlice.actions;
+
+export default categoriesSlice.reducer;
```
Comment (Bondok6) [inline src/redux/categories/categories.js]: - [ ] The same thing here, Please check the comment above ⬆

**Review by Bondok6 — CHANGES_REQUESTED** [review]
## Hi @amejid 🙋‍♂️, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights

- No linter errors ✅
- Your repo is professional ✅
- Professional README file ✅

## Required Changes ♻️
- [ ] Great job adding `MIT.md` file and using the Microverse template 💯, but I noticed you forgot to customize it, So kindly add your name and customize it 👌 

image


- [ ] _Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by amejid** [comment]
Thanks for the feedback, all the issues are addressed in the subsequent commits.

**Review by Bondok6 — APPROVED** [review]
# STATUS: APPROVED :green_circle: :clap: 

## Hi @amejid 🙋‍♂️, 
Your project is complete!  There is nothing else to say other than... Great job 👏 
Congratulations! 🎉 

TO HIGHLIGHT:

- No linter errors ✅
- Great job customizing MIT.md file ✅
- Great Job using Use store, actions, and reducers ✅
- Professional README.md file 💯

Now you can merge this branch and move on :1st_place_medal: 
HAPPY CODING AND GOOD LUCK WITH THE COMMING ONES :crossed_fingers: :four_leaf_clover: 




Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by amejid** [comment]
Thank you for reviewing and approving my code.

**Pull request merged**